backend/clubs/views.py

```diff
@@ -1526,6 +1528,31 @@ def get_queryset(self):
         return Badge.objects.filter(Q(purpose="fair") | Q(label="SAC"))
 
 
+class FavoriteCalendarAPIView(APIView):
+    """
+        get: Return a .ics file of the user's favorite clubs' events.
+    """
+
+    def get(self, request, *args, **kwargs):
+        calendar = ICSCal()
+        calendar.extra.append(ICSParse.ContentLine(name="X-WR-CALNAME", value="Penn Clubs Events"))
+
+        all_events = Event.objects.filter(
+            club__favorite__person__profile__uuid_secret=kwargs["user_secretuuid"]
+        ).distinct()
+        for event in all_events:
+            e = ICSEvent()
+            e.name = event.club.name + ": " + event.name
+            e.begin = event.start_time
+            e.end = event.end_time
+            e.location = event.location
+            e.description = ("" if event.url is None else event.url) + "\n\n" + event.description
```

Comment:
You can simplify this to `(event.url or "")`.

Also, its nice to use format strings. In this case, you can do `"{}\n\n{}".format(event.url or "", event.description)`.